From 6de738c2942e7eba079c7a2fc0e66112289108ee Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Tue, 26 Feb 2008 14:58:17 -0600
Subject: [PATCH] ENGR00066963 ipu: remove double buffer reset on disable

This patch fixes an issue with the SDC channel being set to the
wrong buffer after disabling and re-enabling the channel.

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 drivers/mxc/ipu/ipu_common.c |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/drivers/mxc/ipu/ipu_common.c b/drivers/mxc/ipu/ipu_common.c
index 8de8295..6d6eef0 100644
--- a/drivers/mxc/ipu/ipu_common.c
+++ b/drivers/mxc/ipu/ipu_common.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2005-2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2005-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -1204,9 +1204,6 @@ int32_t ipu_disable_channel(ipu_channel_t channel, bool wait_for_stop)
 	reg = __raw_readl(IDMAC_CHA_EN);
 	__raw_writel(reg & ~chan_mask, IDMAC_CHA_EN);
 
-	/* Reset to buffer 0 */
-	__raw_writel(chan_mask, IPU_CHA_CUR_BUF);
-
 	/* Clear DMA related interrupts */
 	__raw_writel(chan_mask, IPU_INT_STAT_1);
 	__raw_writel(chan_mask, IPU_INT_STAT_2);
-- 
1.5.4.4

